controllers/deployment.go

```diff
@@ -326,6 +338,28 @@ func buildStatefulSetsObject(spec *StatefulSpec) (*reconciler.Object, error) {
 	if err != nil {
 		return nil, err
 	}
+	// For custom volumes and custom volume mounts, we directly pass structs from the spec to bypass the YAML templating logic.
+	k8sObj, ok := obj.Obj.(*k8s.Object)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert object to k8s object")
+	}
+	statefulSetObj, ok := k8sObj.Obj.(*appsv1.StatefulSet)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert meta object to statefulset object")
+	}
+	if err := addVolumeToPodSpec(&statefulSetObj.Spec.Template.Spec, spec.Base.CustomVolumes); err != nil {
+		return nil, err
+	}
+	for ind, _ := range statefulSetObj.Spec.Template.Spec.InitContainers {
```

Comment:
nit: ind => either idx or i. 

Have never seen "ind" as abbreviation for index

Reply:
Fixed